### DISCUSSION TITLE
How does it work the --related cli option?

### DISCUSSION THREAD

**Discussion by giolf** [vitest-dev/vitest#1816] [Q&A] [answered]
I didn't get how the `related` CLI option command works behind the scene.

If i have a `PrimaryButton.vue`file and it's related spec file: `PrimaryButton.spec.ts`,
Why when i run:`npx vitest related src/components/PrimaryButton.vue`
The following `specs`get involved?
```
PrimaryButton.spec.ts
Home.spec.ts
AlertPopup.spec.ts
```
why that? Am i wrong or the `related`option should just run the related spec file of the `PrimaryButton` component?

**Comment by sheremet-va** [comment, marked as answer]
Vitest doesn't have `related` flag, it's a command: `vitest related components/Button.vue`.

**Reply by giolf** [reply]
Sorry my bad: i meant by running `vitest related src/components/PrimaryButton.vue`
It was just a typo, the output is the same... can you explain me why 3 spec tests are runned with:
`npx vitest related src/components/PrimaryButton.vue`

thanks.

**Reply by sheremet-va** [reply]
I don't have your code, so I can give several reasons:

- you are directly importing it inside a test
- you are importing a file that imports PrimaryButton
- you are importing a file that exports PrimaryButton

**Reply by giolf** [reply]
hey @sheremet-va this is exactly what i was meaning: 
what `related` command does behind the scene? I guess it is not simply checking the _related_ spec file of `PrimaryButton` right?
it seems it does much more...

**Reply by sheremet-va** [reply]
Every spec file has a module graph. If vitest finds specified file in a spec module graph, it will run this spec file. Several spec files might import the same file - each of them will be run.

It doesn't matter if spec file imports it directly - it will crawl the whole module graph trying to find it.

**Reply by sheremet-va** [reply]
To put it simply: if spec file imports specified files directly or indirectly, vitest will run this spec file.

**Reply by giolf** [reply]
got it, thank you.
